internal/controller/argocd.go

```diff
@@ -150,6 +150,29 @@ func (clctrl *ClusterController) InitializeArgoCD() error {
 	return nil
 }
 
+func RestartDeployment(ctx context.Context, clientset kubernetes.Interface, namespace string, deployment_name string) error {
+
+	deploy, err := clientset.AppsV1().Deployments(namespace).Get(ctx, deployment_name, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
```

Comment:
It seems you did this on the bottom for the other error, but not for this one, so adding context here.

```suggestion
	if err != nil {
		return fmt.Errorf("unable to get deployment %q: %w", deployment_name, err)
	}
```